File: frontend/src/features/goalBucket/types/goalBucket.ts
export type GoalBucket = {
  goalBucketId: number
  accountId: number
  bucketName: string
  active: boolean
  createdAt: string
  updatedAt: string
}

export type CreateGoalBucketInput = {
  accountId: number
  bucketName: string
  active: boolean
}

File: backend/flowlet/src/main/java/com/example/flowlet/goalbucket/domain/model/GoalBucket.java
package com.example.flowlet.goalbucket.domain.model;

import java.time.LocalDateTime;

public record GoalBucket(
    Long goalBucketId,
    Long accountId,
    String bucketName,
    boolean active,
    LocalDateTime createdAt,
    LocalDateTime updatedAt
) {
}
